### PR TITLE
Fix/allow default value

### PR DESCRIPTION
**Pull request by corentinghigny**
I don't even remember why, but while working on avro compatibility check, we found out strange behaviour by the parser if it encounter null value while parsing String. While trying to fix that, a lot of stuff popped out.

We used that opportunity to fix handling of "null" and implement "default" field value (avro spec).
We also added a parameter "missingDefaultIsNull", that put "null" as "default" if no default is given

More info on why this is needed on https://issues.apache.org/jira/browse/AVRO-1803
(no "null" default means that if key isn't found in the record, the record is invalid)

This parameter is "true" by default for simplicity and avoid breaking compatibility

It **SHOULD** not break anything, UT still works, but it might, has before, in the `fromMap` method of `AvroConverter`, the only line was
```java
record.put(field.name(), convert(field.schema(), data.get(field.name())));
```
this didn't check if the key existed in the record, `Map().get()` returns null if the key doesn't exist

### CODE REVIEW

**Comment by corentinghigny** [comment]
This still needs works : a lot of TODO and FIXME were put in nearly every avro type test
I'd like some help on those because I have deadlines coming around :sweat:

File: src/test/java/org/kestra/task/serdes/avro/converter/ComplexMapTest.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 public class ComplexMapTest {
     static Stream<Arguments> source() {
+        // TODO test null (not "null") in Map
```
Comment (corentinghigny) [inline src/test/java/org/kestra/task/serdes/avro/converter/ComplexMapTest.java]: By that I mean real null, not the string "null"

**Comment by corentinghigny** [comment]
Also, `missingDefaultIsNull` is a lame variable name, some ideas for a better one ?

**Comment by tchiotludo** [comment]
will be a massive breaking change, will clause so far

**Pull request closed without merging**